edgePy/util.py

```diff
@@ -0,0 +1,23 @@
+""" Utilities to support functions and classes """
+import logzero  # type: ignore
+
+
+def _setup_log():
+    """Formats and sets up the logger instance.
+
+    Returns:
+        An instance of a logger.
+    """
+    _log_format = (
+        "%(color)s[%(levelname)s] [%(asctime)s | "
+        "%(module)s - line %(lineno)d]:%(end_color)s %(message)s"
+    )
+    _formatter = logzero.LogFormatter(fmt=_log_format)
+
+    logger = logzero.setup_logger(logfile=None, level=logzero.logging.INFO, formatter=_formatter)
+
+    return logger
+
+
+# Logger instance constant
+LOG = _setup_log()
```

Comment:
No need to setup main logger here.

Every module or script should have the ability to grab a logger with a name appropriate to its function.

I would like to see the following API, this allows for hierarchy of loggers or just keeping a logger instance reserved for scripts separate from that of the main interactive library.

```python
# edgePy.util

def getLogger(name, level=logzero.logging.INFO, formatter=None):
    ...
    logger = logzero.setup_logger(name=name, ...)
    return logger
```

Then in `edgePy` modules:

```python
from edgepy.util import getLogger

logger = getLogger(name=__name__)
```

In edgePy script:

```python
from edgepy.util import getLogger

logger = getLogger(name='script')
```

Reply:
I agree with this. I'm doing something similar in another project, and it works a lot better this way imo. Should be an easy fix.